packages/google_maps_flutter/google_maps_flutter_web/lib/src/marker_clustering.dart

```diff
@@ -0,0 +1,205 @@
+  void addItem(ClusterManagerId clusterManagerId, gmaps.Marker marker) {
+    final MarkerClusterer? markerClusterer =
+        _clusterManagerIdToMarkerClusterer[clusterManagerId];
+    if (markerClusterer != null) {
+      markerClusterer.addMarker(marker, true);
+      markerClusterer.dirty = true;
+    }
+  }
+
+  /// Removes given [gmaps.Marker] from the [MarkerClusterer] with given [ClusterManagerId].
+  void removeItem(ClusterManagerId clusterManagerId, gmaps.Marker? marker) {
+    if (marker != null) {
+      final MarkerClusterer? markerClusterer =
+          _clusterManagerIdToMarkerClusterer[clusterManagerId];
+      if (markerClusterer != null) {
+        markerClusterer.removeMarker(marker, true);
+        markerClusterer.dirty = true;
+      }
+    }
+  }
+
+  // Renders all cluster managers that are marked as dirty.
+  void renderDirty() {
+    _clusterManagerIdToMarkerClusterer.values
+        .where((MarkerClusterer markerClusterer) => markerClusterer.dirty)
+        .forEach((MarkerClusterer markerClusterer) {
+      markerClusterer.dirty = false;
+      markerClusterer.render();
+    });
+  }
+
+  /// Returns list of clusters in [MarkerClusterer] with given [ClusterManagerId].
+  List<Cluster> getClusters(ClusterManagerId clusterManagerId) {
+    final MarkerClusterer? markerClusterer =
+        _clusterManagerIdToMarkerClusterer[clusterManagerId];
+    if (markerClusterer != null) {
+      return markerClusterer.clusters
+          .map((MarkerClustererCluster cluster) =>
+              _convertCluster(clusterManagerId, cluster))
+          .toList();
+    }
+    return <Cluster>[];
+  }
+
+  void _clusterClicked(
+      ClusterManagerId clusterManagerId,
+      gmaps.MapMouseEvent event,
+      MarkerClustererCluster markerClustererCluster,
+      gmaps.GMap map) {
+    if (markerClustererCluster.count > 0 &&
+        markerClustererCluster.bounds != null &&
+        markerClustererCluster.markers != null) {
+      final Cluster cluster =
+          _convertCluster(clusterManagerId, markerClustererCluster);
+      _streamController.add(ClusterTapEvent(mapId, cluster));
+    }
+  }
+
+  /// Converts [MarkerClustererCluster] to [Cluster].
+  Cluster _convertCluster(ClusterManagerId clusterManagerId,
+      MarkerClustererCluster markerClustererCluster) {
+    final LatLng position = _gmLatLngToLatLng(markerClustererCluster.position);
+    final LatLngBounds bounds =
+        _gmLatLngBoundsTolatLngBounds(markerClustererCluster.bounds!);
+
+    final List<MarkerId> markerIds = markerClustererCluster.markers!
+        .map<MarkerId>((gmaps.Marker marker) =>
+            MarkerId(marker.get('markerId')! as String))
+        .toList();
+    return Cluster(
+      clusterManagerId,
+      markerIds,
+      position: position,
+      bounds: bounds,
+    );
+  }
+}
+
+@JS()
+external ClusterClickHandler defaultOnClusterClickHandler;
+
+@JS()
+@anonymous
+class MarkerClustererOptions {
```

Comment:
This needs some of the new `@staticInterop`, coming from `dart:js_interop`, instead of `package:js` (there's a migration to the new interop [incoming](https://github.com/flutter/packages/pull/5254/))

Reply:
I'll check what changes this would need

Reply:
I couldn't find a way to use @staticInterop with objects from the google_maps library because the package does not utilize static interop. Additionally, I did not find a clear method to bypass this issue, as clustering is tightly coupled with the Google Maps javascript objects. If the objects within the google_maps package were implemented using @staticInterop, it should be possible to use these objects as function parameters directly.

Reply:
@ditman as you mentioned the [incoming](https://github.com/flutter/packages/pull/5254/) update that should also update google_maps to version 7.x.x having `package:web `support, might be that moving to use `@staticInterop` is possible after that.

Reply:
Yeah, in fact, I've been told that I need to update the JS-interop on maps (and others) to package:web 0.5.0, which doesn't use `@staticInterop` anymore, and instead extension types, like this: https://github.com/flutter/flutter/pull/143274 (so consider my original comment obsolete, I'll work this week on updating the migration PR!)

Reply:
@ditman given the recent updates and the fact that this PR touches on 5 different plugins, leading often to merge conflicts, (especially now that the changelogs are up to date), I wanted to inquire if there's anything more needed from my side to expedite the finalization of this PR for web?